### PR TITLE
Hardcode terminal total difficulty

### PR DESCRIPTION
**Pull request by mkalinin**
### What's done?
- `TransitionStore` with `compute_terminal_total_difficulty` function are replaced with `TERMINAL_TOTAL_DIFFICULTY` configuration parameter
- Adjustments to different parts of the spec that are corresponding to the above change

### Rationale
- Reduce the complexity of consensus client implementation caused by computed TTD, see https://github.com/ethereum/consensus-specs/issues/2603 for details.
- Hardcoded TTD is a robust solution to prevent execution client from mistakenly following the contentious fork sharing the Mainnet PoW chain in the case of software misconfiguration or failure.

### Dependencies
- [x] The corresponding change should be made to [EIP-3675](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3675.md) -- https://github.com/ethereum/EIPs/pull/3981
- [ ] Should be merged after https://github.com/ethereum/consensus-specs/pull/2581

cc @ajsutton

### CODE REVIEW

**Review by djrtwo — COMMENTED** [review]
a couple of small notes about config. 

otherwise looks good! 
will wait on #2581 to approve so the diff looks cleaner for final review

File: specs/merge/beacon-chain.md [hunk 1]
```diff
@@ -76,6 +77,12 @@ This patch adds transaction execution to the beacon chain as part of the Merge f
 | `GENESIS_GAS_LIMIT` | `uint64(30000000)` (= 30,000,000) |
 | `GENESIS_BASE_FEE_PER_GAS` | `Bytes32('0x00ca9a3b00000000000000000000000000000000000000000000000000000000')` (= 1,000,000,000) |
 
+### Transition settings
+
+| Name | Value |
+| - | - |
+| `TERMINAL_TOTAL_DIFFICULTY` | `uint256(2**256 - 1)` |
```
[the remark this answers is not in the document]
Comment (mkalinin) [reply specs/merge/beacon-chain.md]: Done with adding a placeholder to the config profiles

**Review by djrtwo — APPROVED** [review]
excellent!

**Comment by lsankar4033** [comment]
nice! this is definitely cleaner. re: https://github.com/ethereum/consensus-specs/pull/2599, do you think it makes sense to have a param for `TERMINAL_BLOCK_HASH` as well?

**Comment by mkalinin** [comment]
> nice! this is definitely cleaner. re: #2599, do you think it makes sense to have a param for `TERMINAL_BLOCK_HASH` as well?

Yes, I think this would be the right way to implement override of a terminal block. And if transition ends up with terminal block override then consensus and execution clients will need to be released with hardcoded terminal block hash.

**Pull request merged**